adr/ADR-40.md

```diff
@@ -0,0 +1,72 @@
+# Request Many
+
+| Metadata | Value                 |
+|----------|-----------------------|
+| Date     | 2023-06-26            |
+| Author   | @aricart, @scottf     |
+| Status   | Partially Implemented |
+| Tags     | client                |
+
+## Problem Statement
+Have the client support receiving multiple replies from a single request, instead of limiting the client to the first reply.
+
+## Design
+
+Making the request and handling multiple replies is straightforward. Much like a simplified fetch, the developer 
+will provide some basic strategy information that tell how many messages to wait for and how long to wait for those messages.
+
+* The client doesn't assume success or failure - only that it might receive messages.
+The various options are there to short circuit the length of the wait.
+* The Subsequent Max value allows for waiting for the service with the slowest latency. (scatter gather)
+* The message count allows for waiting for some count of messages or a timer (scatter gather)
+
+## Options
+
+#### Max Wait
+
+The maximum amount of time to wait for responses.
+
+* Responses are accepted until the max time is reached.
+
+#### First Max / Subsequent Max
```

Comment:
IMHO - There's no `fist max` as this is the `max wait`. In conditions where the number of responses is unknown, the `subsequent max` or gap timer becomes a shortcut to reduce the wait the client would have to honor to collect the response. The idea is that when the request is sent, all responders will start sending replies (after whatever the processing time is) so given some time here and there, all responses should be queued up fairly quickly so waiting for the full max_wait is simply increasing latency to complete the operation. By simply waiting for a minimal amount of time for additional time, the latency to collecting the response is reduced.